addons/binding/org.openhab.binding.kodi/src/main/java/org/openhab/binding/kodi/handler/KodiHandler.java

```diff
@@ -0,0 +1,310 @@
+                } else if (command.equals(RefreshType.REFRESH)) {
+                    updateState(CHANNEL_INPUT, UnDefType.UNDEF);
+                }
+                break;
+            case CHANNEL_INPUTTEXT:
+                if (command instanceof StringType) {
+                    connection.inputText(command.toString());
+                    updateState(CHANNEL_INPUTTEXT, UnDefType.UNDEF);
+                } else if (command.equals(RefreshType.REFRESH)) {
+                    updateState(CHANNEL_INPUTTEXT, UnDefType.UNDEF);
+                }
+                break;
+            case CHANNEL_ARTIST:
+                if (command.equals(RefreshType.REFRESH)) {
+                    connection.updatePlayerStatus();
+                }
+                break;
+            case CHANNEL_ALBUM:
+                if (command.equals(RefreshType.REFRESH)) {
+                    connection.updatePlayerStatus();
+                }
+                break;
+            case CHANNEL_TITLE:
+                if (command.equals(RefreshType.REFRESH)) {
+                    connection.updatePlayerStatus();
+                }
+                break;
+            case CHANNEL_MEDIATYPE:
+                if (command.equals(RefreshType.REFRESH)) {
+                    connection.updatePlayerStatus();
+                }
+                break;
+            default:
+                logger.debug("Received unknown channel {}", channelUID.getIdWithoutGroup());
+                break;
+        }
+
+    }
+
+    public void playURI(Command command) {
+        connection.playURI(command.toString());
+    }
+
+    public void playNotificationSoundURI(Command command) {
+        connection.playNotificationSoundURI(command.toString());
+    }
+
+    public PercentType getNotificationSoundVolume() {
+        return new PercentType(connection.getVolume());
+    }
+
+    public void setNotificationSoundVolume(PercentType volume) {
+        connection.setVolume(volume.intValue());
+    }
+
+    @Override
+    public void initialize() {
+
+        try {
+
+            connection.connect(this.getConfig().get(HOST_PARAMETER).toString(),
+                    getIntConfigParameter(PORT_PARAMETER, 9090));
+
+            // Start the connection checker
+            Runnable connectionChecker = new Runnable() {
+                @Override
+                public void run() {
+                    try {
+                        connection.checkConnection();
+                    } catch (Exception ex) {
+                        logger.warn("Exception in check connection to @{}. Cause: {}", connection.getConnectionName(),
+                                ex.getMessage());
+
+                    }
+                }
+            };
+            connectionCheckerFuture = scheduler.scheduleWithFixedDelay(connectionChecker, 1, 10, TimeUnit.SECONDS);
+
+        } catch (Exception e) {
+            logger.error("error during opening connection: {}", e.getMessage());
```

Comment:
reduce to debug level and add e.getMessage() as a third parameter on the updateStatus call instead.